### DISCUSSION TITLE
running wss on port 6001

### DISCUSSION THREAD

**Discussion by ahmadyousefdev** [laravel/valet#1475] [Q&A] [answered]
Hello,
i'm using soketi to send private notifications, but i couldn't make it work. Each time i run the page i get 
```
WebSocket connection to 'wss://backend.test:6001/app/app-key?protocol=7&client=js&version=8.4.0-rc2&flash=false' failed: An SSL error has occurred and a secure connection to the server cannot be made.
```
I secured both the backend and front end apps using valet secure

I tried to edit the Nginx config file for backend.test but didn't find a proper way to do it.

I read that I have to put 
```
location / {

        proxy_pass https://127.0.0.1:6001;  # Proxy WebSocket connections to port 6001
        proxy_read_timeout     60;
        proxy_connect_timeout  60;
        proxy_redirect         off;

        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
```
 under `server {listen 127.0.0.1:443 ssl;` but the `location /` already has `rewrite ^ "/Users/myname/.composer/vendor/laravel/valet/server.php" last;` on it.

when i run `http://backend.test:6001/` I get OK , but when running `https://backend.test:6001/` I get "this site can't be reached"

if anyone knows how to help me run wss on valet it'll be appreciated..

**Comment by drbyte** [comment]
Did you use the valet proxy command to create the site in nginx?

**Reply by ahmadyousefdev** [reply]
@drbyte 
using proxy command converts all the traffic to 6001, I have my app work on the same domain. Do you have any suggestions?

**Reply by ahmadyousefdev** [reply]
I created another proxy for testing `testing-socket.test` and added it as `PUSHER_HOST`
still gets the same error:
```
WebSocket connection to 'wss://testing-socket.test:6001/app/app-key?protocol=7&client=js&version=8.4.0-rc2&flash=false' failed: An SSL error has occurred and a secure connection to the server cannot be made.
```
PS. when visiting https://testing-socket.test/ I get OK..

**Reply by drbyte** [reply]
When you run `valet proxy --secure testing-socket http://127.0.0.1:6001` it will create an nginx site for `https://testing-socket.test` which forwards/proxies to `127.0.0.1:6001`.

```sh
> valet proxy --secure testing-socket http://127.0.0.1:6001
Valet will now proxy [https://testing-socket.test] traffic to [http://127.0.0.1:6001].
```

**Reply by ahmadyousefdev** [reply, marked as answer]
@drbyte Thanks for the info,
I did manage to solve the problem by removing pusher_port= 6001 `PUSHER_PORT=` .. I think that's something!

**Reply by drbyte** [reply]
Yes, that's sensible because the proxy "domain" is listening on https, port 443 ... not 6001.

**Comment by ahmadyousefdev** [comment]
if anyone else needs to make secure websockets work with soketi and laravel valet, I wrote this tutorial that might be useful https://ahmadyousef.me/blog/11
